lib/mail_catcher/web/application.rb

```diff
@@ -5,18 +5,53 @@
 require "uri"
 
 require "sinatra"
-require "skinny"
-
-require "mail_catcher/bus"
+require 'async/websocket/adapters/rack'
 require "mail_catcher/mail"
 
-class Sinatra::Request
-  include Skinny::Helpers
-end
-
 module MailCatcher
   module Web
     class Application < Sinatra::Base
+      def initialize
+        super
+        @connections = Set.new
+        @semaphore = Async::Semaphore.new(512)
```

Comment:
Is this supervising websockets and acting as a message bus? Perhaps that's a separate concern from the core web application, and should be extracted?

Reply:
I removed the semaphores, it was not of use for our application. I think Async-WebSockets do well for our case. thoughts?